### ISSUE TITLE
Breaking reading lists, and increasing db size.

### ISSUE THREAD

**Issue by bareheiny** [mcpierce/comixed#550]
It is possible to "break" the reading list functionality and trigger an increase in database size.

The steps for this are as follows:

Open CX and add comics to a reading list (in my case, the list was already created, and empty).
Open the reading list
Logout before the list is refreshed (Chrome Dev tools status = (pending))

Attempting to log back in fails - the CX process has to be completed stopped and re-started in for a log in to work.

After killing the CX process the database size had increased form 39MB to 310MB.  As per #542, a larger database with more comics on a reading list increased from 970MB to ~3GB.

**Comment by mcpierce** [comment]
I wouldn't think that would trigger such a behavior since loading the reading list is a read, and not a write, operation; i.e., the request is fetching data from the database and not inserting new records.

I would more wonder what other operations were going on prior to the growth and if there was possibly some data that had not been flushed when you first checked the file size.

Regarding logging back in, can you reproduce those steps? And, more importantly, can you try reproducing it but, instead of restarting the application, reload the page and see if you can login?

**Comment by bareheiny** [comment]
After loading my test database, I took a copy for backup purposes - so the size is set with no outstanding transactions.

I'm 99% sure I've refreshed the page and attempted to log back in with the same result - but will try it again.  May not be for a few days or so though.

**Comment by bareheiny** [comment]
Ok, so I added 100 comics to an empty reading list and waited for the operation to complete.
Logged out and killed the CX bat - no noticeable increase in DB size.

I started CX up, and logged in - opened the reading list view, and saw the list showing 100 comics.

Opened the list, and was presented with a "create reading list" window.

I left the window open, and waited for the pending (in Chrome dev tools) to complete - at which stage, the reading list information (name and description) was displayed.

Then I waited for all 100 comic thumbnails to be displayed. 

Once the thumbnails were displayed, I logged out of CX, and killed the CMD window.  The CX database (which as 39MB) is 568MB.


So the "new reading list" issue is down to speed and patience - although being presented with a new list window does cause confusion.

There still doesn't seem to be any obvious reason for the increase in DB size though, and I'm somewhat concerned that the size would increase the longer I left CX running before killing the cmd window.

**Comment by bareheiny** [comment]
Restored the DB, logged in and re-created the 100 comic reading list.

Once the comics had been added, I opened the reading list - and the list information was displayed immediately.

Logged out of CX and closed the cmd window (will be testing delaying this part next), restarted CX and logged back in.

Once the home chart was finished, and all processes had completed, I opend the reading list and selected the 100 comic one - again, I was presented with a new list window.  The 33 process was still pending - so there's some sort of delay happening here.

In the dev tools windows, I see a "lists" process (finisehd in 393 ms, "33" (finished after 2.4 min) and another "lists" process (compelted after 2.4 min) - in that order.

**Comment by bareheiny** [comment]
Delaying closing Chrome, and then the CX cmd window post log-out has no impact - the DB size about 30 mins after logging out of CX is 579 MB.

Again, the only actions taken in CX were to add 100 comics to a reading list, then open the reading list.

No new comics were added, and no other actions occured.  All transctions were completed (aside from the frequent rechecking for new issues).

**Comment by bareheiny** [comment]
Apologies for the mulitple comments - but at this stage, I'm thinking more information is better than less.

I just restored my test database...added 4 new issues, and scrapped two comics.  No other actions were taken.  The DB size is now 63.7MB  

I wasn't expecting 4 new comics and metadata for two to increase the DB by that much.

**Comment by bareheiny** [comment]
Last comment, then I'm leaving this for at least tonight - even an action as simple as adding a Comic Vine API key triggered a DB increase of 11MB.

When I logged out, the only transaction that didn't have a 200 status was the checking for new issues...which appears to happen frequently.

**Comment by bareheiny** [comment]
Can't leave well enough alone.

Changed SQL editors, opened the CX DB then closed the editor...DB was compacted to ~10MB.

Started CX, and it was a lot more responsive.  After closing CX, the DB had increased by ~4MB with no additional information edited.

**Comment by mcpierce** [comment]
When you scrape comics, the backend caches the data retrieved from ComicVine so it doesn't have to go back multiple times. So that can account for at least a part of the database growth.

Another thing to remember when looking at the growth of the database (expanding and shrinking) is that it's not linear to the number of comics. So depending on how H2 stages data to be merged into the tables, how often it does that, etc. the database physical file size can fluctuate around some number.

The last thing I'm wondering is if there might not be a contributing factor to the backed up database that you're restoring for each test? If there's something to it, like H2 data as above, that might not contribute to the weird behavior? When you describe the reading list page always going to the edit after your do a restore seems to indicate to me that there might be something off in that backup that's presenting when you restore it.

**Comment by bareheiny** [comment]
I started with a  fresh DB this time.

Loaded ~2k comics, then (in the same session) created a reading list.

Added 10 comics, no problems viewing the list.
Added another 25, no problems viewing the list.
Added another 50...reading list essentially broke after some time.  Lots of red in the Chrome Dev Tools messages, and the menu names changed to the unfriendly ones (xxxx.menu or something, rather than Menu etc.).


With only those actions, the DB size was around ~800MB, way more than my backup DB - where the backup was taken immediately after completing the ~2k import.
